src/pascy/l2.py

```diff
@@ -13,18 +13,77 @@ class ArpLayer(Layer):
     @staticmethod
     def fields_info():
         # TODO: Implement this :)
-        pass
+        return [ UnsignedShort("hardware_type", 1), UnsignedShort("protocol_type", 0x800),
+                UnsignedByte("hardware_size", 6), UnsignedByte("protocol_size", 4),
+                UnsignedShort("opcode", ArpLayer.OP_IS_AT), 
+                MacAddress("mac_src"), IPAddress("ip_src"),
+                MacAddress("mac_dst", MAC_BROADCAST), 
+                IPAddress("ip_dst", IP_BROADCAST)]
+
+
+class IcmpLayer(Layer):
+    PING_REQUEST = 8
+    PING_REPLY = 0
+
+    NAME = "ICMP"
+
+    @staticmethod
+    def fields_info():
+        return [UnsignedByte('type', IcmpLayer.PING_REPLY), UnsignedByte('code', 0), UnsignedShort('checksum', 0),
+                UnsignedInteger('headers', 0), UnsignedLong('payload_data', 0)]
+
+class IPLayer(Layer):
+    NAME = "IP"
+
+    SHORT_MAX = 0xffff
+    SHORT_SIZE = 16
+    
+    SUB_LAYERS = [
+        [IcmpLayer, "protocol", 1]
+    ]
+    @staticmethod
+    def fields_info():
+        return [UnsignedByte('version_IHL', 0x45), UnsignedByte('service_type', 0), 
+                UnsignedShort('total_length', 0), UnsignedShort('identification', 0),
+                UnsignedShort('flags_frame_offset', 0x4000), UnsignedByte('TTL', 0),
+                UnsignedByte('protocol', 6), UnsignedShort('checksum', 0),
+                IPAddress('src'), IPAddress('dst', IP_BROADCAST)]
+
+    @staticmethod
+    def checksum(msg):
+        s = 0
+          # loop taking 2 characters at a time
+        for i in range(0, len(msg), 2):
+            if i != 10:
+                if (i+1) < len(msg):
+                    a = msg[i]
+                    b = msg[i+1]
+                    s = s + (b+(a << 8))
+                else:
+                    raise "Something Wrong here"
```

Comment:
lol I don't think this is how you `raise` an exception 🙃 
You should `raise` something that inherits from `BaseExcetpion` (or more commonly, `Exception`)